src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedAutoPropAccessorSymbol.cs

```diff
@@ -0,0 +1,205 @@
+                var type = TypeWithAnnotations.Create(comp.GetSpecialType(SpecialType.System_Void));
+
+                if (accessorKind == AccessorKind.Init)
+                {
+                    var initOnlyType = Binder.GetWellKnownType(
+                        comp,
+                        WellKnownType.System_Runtime_CompilerServices_IsExternalInit,
+                        diagnostics,
+                        property.Location);
+
+                    var modifiers = ImmutableArray.Create(CSharpCustomModifier.CreateRequired(initOnlyType));
+                    type = type.WithModifiers(modifiers);
+                }
+                _voidType = type;
+            }
+        }
+
+        public override TypeWithAnnotations ReturnTypeWithAnnotations
+        {
+            get
+            {
+                if (_voidType is object)
+                {
+                    Debug.Assert(MethodKind == MethodKind.PropertySet);
+                    return _voidType.Value;
+                }
+
+                return _property.TypeWithAnnotations;
+            }
+        }
+
+        public override MethodKind MethodKind => _accessorKind == AccessorKind.Get
+            ? MethodKind.PropertyGet
+            : MethodKind.PropertySet;
+
+        internal override bool IsInitOnly => _accessorKind == AccessorKind.Init;
+
+        public override int Arity => 0;
+
+        public override bool IsExtensionMethod => false;
+
+        public override bool HidesBaseMethodsByName => false;
+
+        public override bool IsVararg => false;
+
+        public override bool ReturnsVoid => _voidType is object;
+
+        public override bool IsAsync => false;
+
+        public override RefKind RefKind => RefKind.None;
+
+        public override FlowAnalysisAnnotations ReturnTypeFlowAnalysisAnnotations => FlowAnalysisAnnotations.None;
+
+        public override ImmutableArray<TypeWithAnnotations> TypeArgumentsWithAnnotations => ImmutableArray<TypeWithAnnotations>.Empty;
+
+        public override ImmutableArray<TypeParameterSymbol> TypeParameters => ImmutableArray<TypeParameterSymbol>.Empty;
+
+        public override ImmutableArray<ParameterSymbol> Parameters
+        {
+            get
+            {
+                if (_accessorKind == AccessorKind.Get)
+                {
+                    return ImmutableArray<ParameterSymbol>.Empty;
+                }
+                return ImmutableArray.Create(SynthesizedParameterSymbol.Create(
+                    this,
+                    _property.TypeWithAnnotations,
+                    ordinal: 0,
+                    RefKind.None,
+                    name: ParameterSymbol.ValueParameterName));
+            }
+        }
+
+        public override ImmutableArray<MethodSymbol> ExplicitInterfaceImplementations => ImmutableArray<MethodSymbol>.Empty;
+
+        public override ImmutableArray<CustomModifier> RefCustomModifiers => _property.RefCustomModifiers;
```

Comment:
>_property.RefCustomModifiers [](start = 77, length = 28)

This is definitely wrong for a setter. Also it doesn't look like any any associated property can be ref returning. I think those cannot have setters and it looks like we always create both accessors.